doc/reference-py-repo.rst

```diff
@@ -0,0 +1,78 @@
+..
+  Copyright (C) 2015  Red Hat, Inc.
+
+  This copyrighted material is made available to anyone wishing to use,
+  modify, copy, or redistribute it subject to the terms and conditions of
+  the GNU General Public License v.2, or (at your option) any later version.
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY expressed or implied, including the implied warranties of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+  Public License for more details.  You should have received a copy of the
+  GNU General Public License along with this program; if not, write to the
+  Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+  02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+  source code or documentation are not subject to the GNU General Public
+  License and may only be used or replicated with the express permission of
+  Red Hat, Inc.
+
+************
+Repositories
+************
+
+.. class:: hawkey.Repo
+
+  Instances of :class:`hawkey.Repo` point to metadata of packages that are
+  available to be installed. The metadata are expected to be in the format
+  generated by a program `createrepo <http://createrepo.baseurl.org/>`_.
+
+  .. attribute:: cost
+
+    An integer specifying a relative cost of accessing this repository. If the
+    same package can be downloaded from two or more repositories, the
+    repository with the lowest cost is preferred.
+
+  .. attribute:: filelists_fn
+
+    A valid readable string path to the XML file containing the file and
+    directory listings of the packages if set.
+
+  .. attribute:: name
+
+    A string name of the repository.
+
+  .. attribute:: presto_fn
+
+    A valid readable string path to the XML file containing the delta metadata
+    if set.
+
+  .. attribute:: primary_fn
+
+    A valid readable string path to the XML file containing the primary
+    metadata if set.
+
+  .. attribute:: priority
+
+    An integer priority value of this repository. If there is more than one
+    candidate package for a particular operation, the one from the repository
+    with *the lowest priority* value is picked, possibly despite being less
+    convenient otherwise (e.g. by being a lower version).
```

Comment:
No, both ignores the version. It works this way:
1. find the repositories containing a package that matches the ```spec```
2. then select just those repositories that have the lowest priority
3. then select those priorities that have the lowest cost

I just copy&pasted the description from: http://dnf.readthedocs.org/en/latest/conf_ref.html#repo-options
Libsolv contains in its documentation: "The sub-priority of the repository. This value is compared when the priorities
of two repositories are the same. It is useful to make the library prefer
on-disk repositories to remote ones." (please note that we implement ```cost``` using libsolv's ```subpriority```).
I did not mention ```priority``` in the description of ```cost``` because it's not mention in DNF's documentation. But if you think it's worth, I can.